rotkehlchen/exchanges/okx.py

```diff
@@ -0,0 +1,410 @@
+            except (UnknownAsset, UnsupportedAsset):
+                continue
+
+            try:
+                usd_price = Inquirer().find_usd_price(asset=asset)
+            except RemoteError as e:
+                self.msg_aggregator.add_error(
+                    f'Error processing {self.name} {asset.name} balance result due to inability '
+                    f'to query USD price: {str(e)}. Skipping balance result.',
+                )
+                continue
+
+            try:
+                amount = deserialize_asset_amount(currency_data['availBal'])
+            except DeserializationError as e:
+                self.msg_aggregator.add_error(
+                    f'Error processing {self.name} {asset.name} balance result due to inability '
+                    f'to deserialize asset amount due to {str(e)}. Skipping balance result.',
+                )
+                continue
+
+            assets_balance[asset] += Balance(
+                amount=amount,
+                usd_value=amount * usd_price,
+            )
+
+        return dict(assets_balance), ''
+
+    def query_online_trade_history(
+            self,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> tuple[list[Trade], tuple[Timestamp, Timestamp]]:
+        """
+        https://www.okx.com/docs-v5/en/#rest-api-trade-get-transaction-details-last-3-months
+        """
+        raw_trades = self._paginated_api_query('trades', 'ordId', {
+            'start_ts': start_ts,
+            'end_ts': end_ts,
+        })
+
+        trades: list[Trade] = []
+        for raw_trade in raw_trades:
+            trade = self.trade_from_okx(raw_trade)
+            if trade:
+                trades.append(trade)
+
+        return trades, (start_ts, end_ts)
+
+    def query_online_margin_history(
+            self,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> list[MarginPosition]:
+        return []  # noop for okx
+
+    def query_online_deposits_withdrawals(
+            self,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> list[AssetMovement]:
+        """
+        https://www.okx.com/docs-v5/en/#rest-api-funding-get-deposit-history
+        https://www.okx.com/docs-v5/en/#rest-api-funding-get-withdrawal-history
+        """
+        deposits = self._paginated_api_query('deposits', 'ts')
+        withdrawals = self._paginated_api_query('withdrawals', 'ts')
+
+        movements = []
+        for raw_movement in deposits:
+            movement = self.asset_movement_from_okx(
+                raw_movement,
+                AssetMovementCategory.DEPOSIT,
+            )
+            if movement:
+                movements.append(movement)
+        for raw_movement in withdrawals:
+            movement = self.asset_movement_from_okx(
+                raw_movement,
+                AssetMovementCategory.WITHDRAWAL,
+            )
+            if movement:
```

Comment:
we try to be as explicit as possible with the expressions

```suggestion
            if movement is not None:
```